### PR TITLE
[+] BO: Show import button only if employee has import privilege

### PR DESCRIPTION
**Pull request by guestisp**
Import button in list header is now shown only if logged employee has access to AdminImport controller. This prevent a broken link.

### CODE REVIEW

**Comment by guestisp** [comment]
Probably we should check for `add` capability but actually PS allow importing even with just `view` capability on `AdminImport` controller

**Comment by guestisp** [comment]
News for this? Standard PS behavious is buggy and makes a broken link

File: classes/controller/AdminController.php [hunk 1]
```diff
@@ -297,6 +297,9 @@ class AdminControllerCore extends Controller
 
 	protected $logged_on_addons = false;
 
+	/** @var bool if logged employee has access to AdminImport */
+	protected $canImport = false;
```
Comment (jeromenadaud) [inline classes/controller/AdminController.php]: Hi,

In PrestaShop variable must be in snakeCase, please check http://doc.prestashop.com/display/PS16/Coding+Standards

Best regards.

File: classes/controller/AdminController.php [hunk 2]
```diff
@@ -417,9 +420,15 @@ public function __construct()
 		else
 			$this->context->mode = Context::MODE_STD;
 
+		//* Check if logged employee has access to AdminImport controller */
+		$importAccess = Profile::getProfileAccess($this->context->employee->id_profile, Tab::getIdFromClassName('AdminImport'));
```
Comment (jeromenadaud) [inline classes/controller/AdminController.php]: norms !

File: controllers/admin/AdminAddressesController.php [hunk 3]
```diff
@@ -83,10 +83,11 @@ public function initToolbar()
 	{
 		parent::initToolbar();
 		if (!$this->display)
-			$this->toolbar_btn['import'] = array(
-				'href' => $this->context->link->getAdminLink('AdminImport', true).'&import_type=addresses',
-				'desc' => $this->l('Import')
-			);
+			if ( $this->canImport )
```
Comment (jeromenadaud) [inline controllers/admin/AdminAddressesController.php]: norms ! no extra space after and before parenthesis. Thanks

File: controllers/admin/AdminSuppliersController.php [hunk 4]
```diff
@@ -334,11 +334,13 @@ public function initToolbar()
 	{
 		parent::initToolbar();
 		$this->addPageHeaderToolBarModulesListButton();
-		if (empty($this->display))
-			$this->toolbar_btn['import'] = array(
-				'href' => $this->context->link->getAdminLink('AdminImport', true).'&import_type=suppliers',
-				'desc' => $this->l('Import')
-			);
+		if (empty($this->display)) {
```
Comment (jeromenadaud) [inline controllers/admin/AdminSuppliersController.php]: norms !
accolade must be on a new line, thanks.

File: classes/controller/AdminController.php (hunk 1 above)
Comment (guestisp) [inline classes/controller/AdminController.php]: Are you referring to $logged_on_addons ?
That's not mine but i'll fix

Comment (jeromenadaud) [inline classes/controller/AdminController.php]: Hi, I referrer to $canImport, PrestaShop norms want $can_import;

Comment (guestisp) [inline classes/controller/AdminController.php]: Sorry, I understood. You wrote "snakeCase" using "camelCase". (snake_case / camelCase)

**Comment by guestisp** [comment]
Should be ok now

**Comment by jeromenadaud** [comment]
Hi, Thank you for your contribution

Best regards.

**Pull request merged**